sound/soc/sof/intel/intel-client.c

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-2.0-only
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+//
+
+#include <linux/module.h>
+#include "../sof-priv.h"
+#include "../sof-client.h"
+#include "intel-client.h"
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST_CLIENT)
+int intel_register_ipc_test_clients(struct snd_sof_dev *sdev)
+{
```

Comment:
Is this really Intel specific?
Can the core create the devices for the IPC flood test?

In sound/soc/sof/Kconfig we could have:

```
config SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST_CLIENT
	tristate "SOF enable IPC flood test client"
	depends on SND_SOC_SOF_CLIENT
	help
	  This option enables a separate client device for IPC flood test
	  which can be used to flood the DSP with test IPCs and gather stats
	  about response times.
	  Say Y if you want to enable IPC flood test.
	  If unsure, select "N".

if SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST_CLIENT
config SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST_CLIENT_NUM
	int "Number of IPC flood test clients"
	range 1 32
	default 2
	help
	  Select the number of IPC flood test clients to be created.
endif
```
The core would create SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST_CLIENT_NUM auxdev for IPC flood testing and the same config can be used on all platform with consistency.

Reply:
Good question @ujfalusi 

@ranj063 is IPC flood test only for Intel? The probes are certainly Intel-specific, but the IPC tests seem like generic-enough that they could also be enabled on ARM, no?

Reply:
Looking at the sof-ipc-test-client.c, it is just sending messages with
`hdr.cmd = SOF_IPC_GLB_TEST_MSG | SOF_IPC_TEST_IPC_FLOOD;`
in a loop. More instance is there, more loops are running parallel, testing the Linux side of locking I suppose.

From the looks I don't see how this fits on the auxbus, it is not a feature of the DSP or the hardware, it is a SW test on the Linux side.
I think this might get questioned upstream....